### PR TITLE
Initial rename sunkit-spex

### PR DESCRIPTION
**Pull request by samaloney**
Rename package to `sunkit-spex` and module to `sunlit_spex`

- [x] Run tests
- [x] Check doc build
- [x] Check example notebooks
      * don't have all the example data
      * some errors show in after certain plots but plots are shown so looks unrelated 

 *These can really only be done after this is merged*
- [ ] Rename repo
- [ ] Update CI
- [ ] Update RTD

### CODE REVIEW

**Comment by samaloney** [comment]
Mac tests pass, the fail is caused by an issue submitting the codecov report.

File: sunkit_spec/sunxspex_fitting/tests/test_fitter.py
Comment (hayesla) [inline sunkit_spec/sunxspex_fitting/tests/test_fitter.py:7]: this may be confusing, and an issue with namespace?

we should probably rename the class to something else - any suggestions?

File: sunkit_spec/sunxspex_fitting/fitter.py
Comment (hayesla) [inline sunkit_spec/sunxspex_fitting/fitter.py:62]: this class name probably shouldnt be the same name as the package

File: sunkit_spec/sunxspex_fitting/tests/test_fitter.py
Comment (hayesla) [inline sunkit_spec/sunxspex_fitting/tests/test_fitter.py:54]: same here etc

**Review by hayesla — COMMENTED** [review]
Thansk for doing this @samaloney 

Looks good, apart from the class name - we should rename that to something else than the package name

**Comment by settwi** [comment]
Can we close the bug fixes/photon power law before merging this please?

File: sunkit_spec/sunxspex_fitting/tests/test_fitter.py
Comment (KriSun95) [reply sunkit_spec/sunxspex_fitting/tests/test_fitter.py:7]: If you mean `sunxspex_fitting` then I agree! Maybe even just `fitting` would do?

File: sunkit_spec/sunxspex_fitting/fitter.py
Comment (KriSun95) [reply sunkit_spec/sunxspex_fitting/fitter.py:62]: I agree with this as well but I'm not sure what a good replacement would be. I can only think of something like `Fitter` but I'm not sure that's much better. Also, just to double check, was the name that won the poll `sunkit-spec` or `sunkit-spex`?

**Comment by KriSun95** [comment]
> Can we close the bug fixes/photon power law before merging this please?

You reminded me of a conversation with @DanRyanIrish at SPD that this was OK to be accepted from his point of view so, before any larger changes are made to the repo, I'll merge the PR in question. Any other changes down the line can be addressed in the refactoring.

File: sunkit_spec/sunxspex_fitting/tests/test_fitter.py
Comment (samaloney) [reply sunkit_spec/sunxspex_fitting/tests/test_fitter.py]: Yea I don't want to change too much in his PR also kinda want to leave `sunlit_spex.fitting` free for the refactor.

File: sunkit_spec/sunxspex_fitting/fitter.py
Comment (samaloney) [reply sunkit_spec/sunxspex_fitting/fitter.py]: I've renamed it `Fitter` for the moment again not a final name just temporary work around.

**Comment by samaloney** [comment]
Ok think this is all good to go now, once this is merged will prob have to be a small tidy up PR after we rename the GitHub repo if we want to do that, I presume we do?

**Comment by settwi** [comment]
Thank you @samaloney! Looks comprehensive.
After merging this it might be worthwhile to rename the CT/CI stuff too, and fix its public key issues. Do the "other" test issues mean they just failed to run? Are we sure things will not severely break after the merge?
- WS

File: sunkit_spex/sunxspex_fitting/data_loader.py
Comment (KriSun95) [inline sunkit_spex/sunxspex_fitting/data_loader.py:1]: Should `sunkit_spex` replace `Fitter` in the comments and doc strings?

File: sunkit_spex/sunxspex_fitting/fitter.py
Comment (KriSun95) [inline sunkit_spex/sunxspex_fitting/fitter.py:69]: `sunkit_spex`

File: sunkit_spex/sunxspex_fitting/fitter.py
Comment (KriSun95) [inline sunkit_spex/sunxspex_fitting/fitter.py:780]: `sunkit_spex`

File: sunkit_spex/thermal.py
Comment (KriSun95) [inline sunkit_spex/thermal.py:749]: `sunkit_spex`

File: tox.ini
Comment (KriSun95) [inline tox.ini:45]: Should `Fitter` be `sunkit_spex` here?

**Review by KriSun95 — COMMENTED** [review]
I've pointed out what I think might be some typos in the replacement of `sunxspex` which looks like it was sometimes replaced with `Fitter` instead of `sunkit_spex` when referring to the whole package. Apologies since that's my fault by calling the fitter the same name as the package.

**Comment by samaloney** [comment]
@settwi unfortunately our test coverage isn't great so other than running the test we have and checking the example notebooks work I'm not sure how to know if anything will break.

@KriSun95 ah thanks I missed some of those especially in the comments, I think I have them all now 🔍

**Review by KriSun95 — APPROVED** [review]

**Comment by KriSun95** [comment]
Are we close to merging this? Do we just need another approver?

**Comment by samaloney** [comment]
Yea or I can just merge?

**Comment by settwi** [comment]
Merge it!
Or as the Americans are more likely to say (do): PULL THE TRIGGER!

**Pull request merged**
